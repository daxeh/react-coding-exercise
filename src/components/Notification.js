import React from 'react'
import injectSheet from 'react-jss'
import theme from '../style/theme'
import classNames from 'classnames'

const Notification = ({ classes, code, message, clickLabel, clickHandler }) => {
  return (
    <div className={classNames(classes.container)}>
      <div className={classes.content}>
        <p>{message} #{code}</p>
        {clickHandler && (<div className={classes.bottom} onClick={clickHandler}>{clickLabel}</div>)}
      </div>
    </div>
  )
}

export default injectSheet({
  container: {
    top: 0,
    left: 0,
    right: 0,
    width: 500,
    padding: 0,
    color: '#fff',
    height: 100,
    margin: [0, 'auto'],
    display: props => props.message ? 'block' : 'none',
    'z-index': 10,
    position: 'fixed',
    'background-color': '#000',
    opacity: '0.8'
  },
  content: {
    padding: 20,
    height: '100%',
    position: 'relative',
    'flex-direction': 'column',
    overflow: 'hidden'
  },
  bottom: {
    height: 22,
    width: 60,
    right: 0,
    color: theme.colors.primary,
    cursor: 'pointer',
    'text-align': 'center',
    position: 'absolute',
    textTransform: 'uppercase'
  },
  hidden: {
    display: 'none'
  }
})(Notification)
