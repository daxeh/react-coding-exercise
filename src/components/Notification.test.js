import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import Notification from './Notification'

describe('Notification', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Notification />, div)
  })

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Notification debug />)
    expect(component).toMatchSnapshot()
  })
})
