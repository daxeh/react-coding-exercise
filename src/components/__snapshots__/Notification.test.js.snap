// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Notification)
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "bottom": "Notification-bottom-0-1-3",
        "container": "Notification-container-0-1-1 Notification-container-0-1-6",
        "content": "Notification-content-0-1-2",
        "hidden": "Notification-hidden-0-1-4",
      },
      "debug": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "bottom": "Notification-bottom-0-1-3",
          "container": "Notification-container-0-1-1 Notification-container-0-1-6",
          "content": "Notification-content-0-1-2",
          "hidden": "Notification-hidden-0-1-4",
        },
        "debug": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
